universal.md

```diff
@@ -0,0 +1,423 @@
+
+- **`components`** *array of components*
+
+  An array of OPTIONAL components of the object, such as layers and config for an image. See below for details, this is parsed to extract the `blob` and `reference` links for gc.
+
+- **`annotations`** *string-string map*
+
+  Standard OCI annotations as per existing standards, this is metadata that may have common fields across different objects. Arguably this should be a map of string to string array so that annotation keys can be repeated, so they map better to other structures (such as http headers) that allow repetition.
+
+- **any other data** *unspecified*
+
+  Any other fields with any other structure. The client can use this data in any way that is appropriate, but generic tooling will not usually be able to do anything with it as there are no standards.
+
+The `components` of an object correspond to individual parts, potentially of multiple types, for example an image has a config and some layers. The parts that are needed to extract generic data are specified but again any type specific data can be added.
+
+- **`rtype`** *string*
+
+  The reference type of the component, which MUST be `blob`, `reference` or not specified if the component has neither a blob or reference link. This would be for a case where it is a component that does not have a descriptor, eg it refers to something by tag not hash so is not tracked as a reference. I am also wondering if we should add `data` for a "data URI" type reference where there would be a blob but it is inlined into the object instead, useful for small unique objects to save an external blob lookup, but the client can treat them exactly like blobs.
+
+- **`descriptor`** *descriptor*
+
+  Standard OCI descriptor. If the component is not a `blob` or `reference` this will be ignored but the client can use it, eg for a foreign layer or for some other reference type that still wants to verify a hash, such as a bridge to another system.
+
+- **`ctype`** *string*
+
+  The OPTIONAL component type, for an image this might be `org.oci.layer` or `org.oci.config`. These benefit from standardisation, as lots of artifacts use configs.
+
+- **any other data** *unspecified*
+
+  Any other fields with any other structure. The client can use this data in any way that is appropriate, but generic tooling will not usually be able to do anything with it as there are no standards.
+
+Now let us look at how we map various existing and proposed types to this manifest format.
+
+The simplest type is the Pointer that simply points to another object. This looks useless, but if it is signed it can act as a detached signature for the item that it points at.
```

Comment:
In this example, where would the signature be stored?

Reply:
In the JWS signature, not shown here (see previous comment)